R/sim_network_bp.R

```diff
@@ -92,9 +93,16 @@
           infect <- stats::rbinom(n = contacts[i], size = 1, prob = prob_infect)
           infected[vec_idx] <- as.numeric(infect)
 
-          # add delay time
-          time[vec_idx] <- contact_interval(length(vec_idx)) +
-            time[ancestor_idx[i]]
+          # compute infectious period for ancestor
+          contact_infect_period <- infect_period(1)
+
+          # assume contacts are evenly distributed across the infectious period
+          contact_times <- stats::runif(
+            n = length(vec_idx),
+            min = 0,
+            max = contact_infect_period
+          )
+          time[vec_idx] <- contact_times + time[ancestor_idx[i]]
```

Comment:
We are assuming here that people will be infected if they are in contact with an infected contact during their infectious period right?

I wonder if `contact_times` should be renamed to something like `contact_infection_times` so that it's easier to see how line 105 adds the same conceptual times.

Reply:
My understanding was that it was more that we only care about tracking timing of contact for those contacts who end up infected? And hence only sample contact times from infectious period? But agree relabelling as `contact_infection_times` might be clearer if it is only defined here and doesn't impact rest of the function.

Reply:
> We are assuming here that people will be infected if they are in contact with an infected contact during their infectious period right?

That is not how it is currently coded. The contact time samples the timing that a contact is made, independent of whether they become infected, which is determined above by binomially sampling infections from contacts by: `infect <- stats::rbinom(n = contacts[i], size = 1, prob = prob_infect)`

> I wonder if contact_times should be renamed to something like contact_infection_times so that it's easier to see how line 105 adds the same conceptual times.

Given the explanation above I think sticking with `contact_times` as it is the time an infected individual has a contact with another individual makes more sense. In the case when a contact becomes infected the contact time is the time they become infected, but in the case where the contact is not infected it is just a contact time.

The model is slightly unrealistic because we only record contacts that are made while a case is infectious, but this also makes sense when we consider we only record one level of contacts and not contacts of contacts.

Reply:
I think we're talking about the same thing, just in slightly different terms. As I understand it, the model stores the number of contacts made for each infected person, but only the timing of contact for those infected (i.e. equivalent to infection times)? Have added a suggestion above to have a comment in the code to clarify.

Reply:
Ah OK, thanks. Still, that implementation seems reasonable, as those recorded contacts-made-while-infectious are the ones contact tracers would typically focus on.